Tools/TestWebKitAPI/Tests/WebKitGLib/TestUIClient.cpp

```diff
@@ -253,6 +253,30 @@ class UIClientTest: public WebViewTest {
         g_main_loop_quit(test->m_mainLoop);
     }
 
+    static gboolean queryPermissionStateCallback(WebKitWebView*, WebKitPermissionStateQuery* query, UIClientTest* test)
+    {
+        WebKitPermissionState state = WEBKIT_PERMISSION_STATE_PROMPT;
+
+        if (!g_strcmp0(test->m_expectedQueryPermissionReply, "granted"))
+            state = WEBKIT_PERMISSION_STATE_GRANTED;
+        if (!g_strcmp0(test->m_expectedQueryPermissionReply, "denied"))
+            state = WEBKIT_PERMISSION_STATE_DENIED;
```

Comment:
I made a comment that I can't find, so I guess I didn't press the comment button or it was lost somehow. Anyway, I said that we should test the case of not handling the signal. I would add a new expectation, or example "prompt-default" that simply returns `FALSE` without calling the finish method.